src/main/java/com/google/cloud/tools/app/impl/cloudsdk/internal/process/DefaultProcessRunner.java

```diff
@@ -79,94 +97,64 @@ public void run(String[] command) throws ProcessRunnerException {
-   *
-   * @param stdOutLineListener Can be null.
-   */
-  public void setStdOutLineListener(ProcessOutputLineListener stdOutLineListener) {
-    this.stdOutLineListener = stdOutLineListener;
-  }
-
-  /**
-   * Set the listener for standard error output of the subprocess. Note that this will not work if
-   * you set inheritIO.
-   *
-   * @param stdErrLineListener Can be null.
-   */
-  public void setStdErrLineListener(ProcessOutputLineListener stdErrLineListener) {
-    this.stdErrLineListener = stdErrLineListener;
-  }
-
-  /**
-   * Sets the subprocess exit listener for collecting the exit code of the subprocess.
-   *
-   * @param exitListener Can be nul.
-   */
-  public void setExitListener(ProcessExitListener exitListener) {
-    this.exitListener = exitListener;
-  }
-
-  /**
-   * @return The process that is currently executing or was the last one to be started using the run
-   *     method.
+   * The process that is currently executing or was the last one to be started using the run
+   * method.
    */
   public Process getProcess() {
     return this.process;
   }
 
   private void handleErrOut(final Process process) {
-    if (stdErrLineListener != null) {
-      final Scanner stdErr = new Scanner(process.getErrorStream());
-      Thread stdErrThread = new Thread("standard-err") {
-        public void run() {
-          while (stdErr.hasNextLine() && !Thread.interrupted()) {
-            String line = stdErr.nextLine();
-            stdErrLineListener.outputLine(line);
-          }
+    final Scanner stdErr = new Scanner(process.getErrorStream());
+    Thread stdErrThread = new Thread("standard-err") {
+      public void run() {
+        while (stdErr.hasNextLine() && !Thread.interrupted()) {
+          String line = stdErr.nextLine();
+          consumeLine(line, true);
         }
-      };
-      stdErrThread.setDaemon(true);
-      stdErrThread.start();
-    }
+      }
+    };
+    stdErrThread.setDaemon(true);
+    stdErrThread.start();
   }
 
   private void handleStdOut(final Process process) {
-    if (stdOutLineListener != null) {
-      final Scanner stdOut = new Scanner(process.getInputStream());
-      Thread stdOutThread = new Thread("standard-out") {
-        public void run() {
-          while (stdOut.hasNextLine() && !Thread.interrupted()) {
-            String line = stdOut.nextLine();
-            stdOutLineListener.outputLine(line);
-          }
+    final Scanner stdOut = new Scanner(process.getInputStream());
+    Thread stdOutThread = new Thread("standard-out") {
+      public void run() {
+        while (stdOut.hasNextLine() && !Thread.interrupted()) {
+          String line = stdOut.nextLine();
+          consumeLine(line, false);
         }
-      };
-      stdOutThread.setDaemon(true);
-      stdOutThread.start();
+      }
+    };
+    stdOutThread.setDaemon(true);
+    stdOutThread.start();
+  }
+
+
+  private void consumeLine(String line, boolean errorStream) {
+    if (errorStream) {
```

Comment:
This method seems to have lost its value, not doing much anymore after switching on the value of `errorstream`. Can we consume the line in the handleX methods now?